subjects/traversal.py

```diff
@@ -63,8 +64,11 @@ def create_browse(self, subject, subject_id, subject_name):
 
         if subject_id != '' and subject_id is not None and subject_name == '':
             if subject_id != 0:
-                # Grabs the row object of the given node
-                current_subject = subject.objects.get(id__exact=subject_id)
+                try:
+                    # Grabs the row object of the given node
+                    current_subject = subject.objects.get(id__exact=subject_id)
+                except subject.DoesNotExist:
+                    raise Http404('The subject does not exist.') 
```

Comment:
We could use [`get_object_or_404`](https://docs.djangoproject.com/en/2.2/topics/http/shortcuts/#get-object-or-404), though this works too. Perhaps we should also add a test in test_views.TestBrowseSubjects for a non-existent id.